File: DemoApplication/ObjcDemoApp/ObjcDemoApp/SetUserIdViewController.m
//
//  setUserIdViewController.m
//  HelloWorld
//
//  Created by Elkana Orbach on 17/01/2018.
//  Copyright © 2018 Optimove. All rights reserved.
//

#import "SetUserIdViewController.h"

@implementation SetUserIdViewController
- (IBAction)userPressOnSend {
    NSString* name = [self.userIdTextField text];
    [Optimove.shared setUserId:name];
}

-(void) viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    [Optimove.shared setScreenVisitWithScreenPathArray:@[@"main_screen",@"set user id"]  screenTitle:@"set user id" screenCategory:nil];
}
@end

File: DemoApplication/SwiftDemoApp/SwiftDemoApp/SetUserIDViewController.swift

import UIKit
import OptimoveSDK

class SetUserIDViewController: UIViewController
{
    @IBOutlet weak var userIDTextField: UITextField!
    
    @IBAction func pressSetUserID(_ sender: UIButton)
    {
        if let text = userIDTextField.text
        {
            Optimove.shared.setUserId(text)
        }
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        Optimove.shared.setScreenVisit(screenPathArray: ["main_screen","login"], screenTitle: "login")
    }
}
